### PR TITLE
move mysql creds to compose env variables

### LINKED ISSUE #75: Compose file should not use notarymysql which has hardcoded passwords
The default suggested deployment shouldn't have the dockercon demo hardcoded stuff, the database should be created by the user then the schema can be initialized by the server itself if it knows it's empty

### PR DESCRIPTION
**Pull request by jfrazelle**
closes #75

### CODE REVIEW

File: cmd/notary-server/config.json [hunk 1]
```diff
@@ -13,6 +13,6 @@
 	},
 	"storage": {
 		"backend": "mysql",
-		"db_url": "dockercondemo:dockercondemo@tcp(notarymysql:3306)/dockercondemo"
+		"db_url": "gordontheturtle:dockerrulez@tcp(notarymysql:3306)/development"
```
Comment (RichardScothern) [inline cmd/notary-server/config.json]: Should this line be referencing DB_USER et al?

Comment (jfrazelle) [inline cmd/notary-server/config.json]: i believe this is the right format user:password@

Comment (RichardScothern) [inline cmd/notary-server/config.json]: The format is fine, but the value are hard-coded.  What I mean was the connection string should reference the env variables like : "${DB_USER}:${DB_PASS}@"

Comment (jfrazelle) [inline cmd/notary-server/config.json]: Will that work w cobra we whatever lib you use I'm not familiar

Comment (diogomonica) [inline cmd/notary-server/config.json]: I agree, this should reference the env variables, and they should be setup before this container runs.

Comment (jfrazelle) [inline cmd/notary-server/config.json]: does that work w cobra

Comment (jfrazelle) [inline cmd/notary-server/config.json]: ive never used that lib lol

Comment (RichardScothern) [inline cmd/notary-server/config.json]: I have no idea what Cobra is :)  

This thing will be deployed with Ansible, until infra switches to docker-compose, so changes to this file are for now immaterial

Comment (jfrazelle) [inline cmd/notary-server/config.json]: oooooo i see, but in the container i see this being a problem maybe...

Comment (jfrazelle) [inline cmd/notary-server/config.json]: ok ya i think it works

**Comment by jfrazelle** [comment]
buellerrrrrr

ferris

**Comment by jfrazelle** [comment]
hi!

**Comment by RichardScothern** [comment]
ping @endophage

**Comment by thaJeztah** [comment]
https://github.com/docker/docker/pull/9176#issuecomment-99542089 🙊👼

**Comment by diogomonica** [comment]
@thaJeztah the irony is not lost on us :)

The major difference is between doing it/condoning it for everyone using docker.

**Comment by thaJeztah** [comment]
@diogomonica I know, sorry, couldn't contain myself :smile:

(change LGTM)

**Comment by endophage** [comment]
@RichardScothern Aaron is using a different compose file to deploy (he builds his own images from those we push to hub and deploys those). We want the files as they exist in the repo to be runnable locally so contributors can get up and running easily. 

With that in mind, we probably need to revert the changes for ansible.

**Comment by RichardScothern** [comment]
@endophage I'm aware of this :)  What do you want to do with this PR? :cake:

**Comment by endophage** [comment]
@RichardScothern you personally, nothing, just addressing your earlier comments.

**Comment by jfrazelle** [comment]
so can i go back to what i had here originally ;) https://github.com/docker/notary/pull/76#discussion-diff-34908672

**Comment by jfrazelle** [comment]
updated

File: cmd/notary-server/config.json [hunk 2]
```diff
@@ -12,6 +12,6 @@
 	},
 	"storage": {
 		"backend": "mysql",
-		"db_url": "dockercondemo:dockercondemo@tcp(notarymysql:3306)/dockercondemo"
+        "db_url": "gordontheturtle:dockerrulez@tcp(notarymysql:3306)/development"
```
Comment (thaJeztah) [inline cmd/notary-server/config.json]: Indentation looks off here

Comment (jfrazelle) [inline cmd/notary-server/config.json]: having a rough day over here haha

Comment (jfrazelle) [inline cmd/notary-server/config.json]: lol

Comment (thaJeztah) [inline cmd/notary-server/config.json]: I saw, no worries :+1:

Comment (jfrazelle) [inline cmd/notary-server/config.json]: retabbed the whole thing, it was def mixed tabs and spaces :)

Comment (thaJeztah) [inline cmd/notary-server/config.json]: Thx, looks good now!

**Comment by diogomonica** [comment]
does docker-compose up work with this config? (as in, are we able to point notary to it and it works?)

**Comment by jfrazelle** [comment]
yessir

**Comment by jfrazelle** [comment]
but you may want to verify for yourself

**Comment by jfrazelle** [comment]
rebased..

**Comment by jfrazelle** [comment]
come on people

**Comment by endophage** [comment]
Now that I've been playing with Ansible, there's basically no difference in whether we include the variables in a config file or as ENV variables, they will get defined in the same place then the `reefer` tool is responsible for either passing them to the process we run or placeholding them into a config . However for local dev, it seems much easier to include them in a config. Thoughts?

**Comment by endophage** [comment]
@jfrazelle going to close this as it'll be superceded by this PR https://github.com/docker/notary/pull/185

That PR will allow trust to take full control over deployments once I finish the integration.

**Pull request closed without merging**